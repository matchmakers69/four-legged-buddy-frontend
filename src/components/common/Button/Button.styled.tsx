import styled from "styled-components";

export const Button = styled.button`
  cursor: pointer;
  box-shadow: none;
  text-shadow: none;
  border-radius: 4px;
  height: 40px;
  padding: 1.5rem 2.2rem;
  margin: 0;
  display: inline-flex;
  min-width: 12rem;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  transition: all 0.3s ease;
  outline: transparent;
  font-size: ${({ theme }) => theme.fontSizes.sm};
  border: none;

  &:not(:last-child) {
    margin-right: 16px;
  }

  &.btn--danger {
    background: ${({ theme }) => theme.colors.red};
    color: ${({ theme }) => theme.colors.powderWhite};
  }

  &.btn--secondary {
    background: ${({ theme }) => theme.colors.lightBrown};
    color: ${({ theme }) => theme.colors.powderWhite};
  }

  &:disabled {
    opacity: 0.3;
    pointer-events: none;
  }
  &:focus {
    color: ${({ theme }) => theme.colors.powderWhite};
    box-shadow: 0px 0px 8px 0px #e9f8ff;
    outline: ${({ theme }) => `1px solid ${theme.colors.lightBlue}`};
  }
`;
