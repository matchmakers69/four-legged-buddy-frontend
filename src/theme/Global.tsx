import { createGlobalStyle } from "styled-components";
import { Theme } from "./theme";

interface IGlobalProps {
  theme: Theme;
}

export const GlobalStyle = createGlobalStyle<IGlobalProps>`
 *, *::before, *::after {
    box-sizing: border-box;
  }
 html {
    font-size: 62.5%;
  }
  
  body {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    overflow-x: hidden;
    line-height: 1.6;
    font: 400 1.6rem 'Roboto', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    background: ${({ theme }) => theme.colors.bodyYellow};
    color: ${({ theme }) => theme.colors.text};
  }
  a {
    text-decoration: none;
  }

  ul, ol {
    padding: 0;
    margin:0;
    list-style: none;
  }
`;
