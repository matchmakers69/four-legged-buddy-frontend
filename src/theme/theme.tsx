import baseStyled, { ThemedStyledInterface, DefaultTheme } from "styled-components";

export const theme: DefaultTheme = {
  title: "theme",
  colors: {
    bodyYellow: "#eae2c1",
    footerGreen: "#18280b",
    powderWhite: "#ffffff",
    persianGreen: "#daee73",
    text: "#010000",
    yellow: "#ead96d",
    grey100: "#e8e8e0",
    lightBrown: "#ecdcc2",
    red: "#c0362b",
    logoLightGreen: "#739c3e",
    logoDarkGreen: "#1c2e0d",
    mustard: "#bb8918",
    lightGreen: "#e5daac",
    titleGreen: "#2b331f",
    lightBlue: "#2193a2",
  },

  fontSizes: {
    xs: "1.4rem",
    sm: "1.6rem",
    md: "2rem",
    large: "3rem",
  },
  fontFamily: {
    primary: "Cabin",
    secondary: "Roboto Slab",
  },
  weight: {
    thin: 300,
    regular: 400,
    medium: 500,
    semiBold: 600,
    bold: 700,
  },
};
export type Theme = typeof theme;
export const styled = baseStyled as ThemedStyledInterface<Theme>;
