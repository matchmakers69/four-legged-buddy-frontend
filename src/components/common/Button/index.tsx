import { FC, ReactNode } from "react";
import cx from "classnames";
import * as S from "./Button.styled";

interface IProps {
  disable?: boolean;
  fullWidth?: boolean;
  variant: string;
  className?: string;
  onClick?: () => void;
  type: "button" | "submit" | "reset";
  children: ReactNode;
}

const Button: FC<IProps> = ({
  disable = false,
  onClick = () => null,
  type,
  fullWidth = false,
  variant,
  className = "",
  children,
}) => {
  const buttonClasses = cx("btn", `btn--${variant}`, className, {
    "w-full justify-center": fullWidth,
  });
  return (
    <S.Button className={buttonClasses} type={type} disabled={disable} onClick={onClick} data-testid="button">
      {children}
    </S.Button>
  );
};

export default Button;
