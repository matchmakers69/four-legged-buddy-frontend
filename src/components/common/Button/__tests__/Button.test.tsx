/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { ThemeProvider as StyledThemeProvider } from "styled-components";
import { theme } from "src/theme/theme";
import Button from "..";

const children = <div /> || "";

describe("<Button />", () => {
  it("sholud create a snapshot", () => {
    expect(
      renderer
        .create(
          <StyledThemeProvider theme={theme}>
            <Button variant="filled" onClick={() => undefined} type="button">
              {children}
            </Button>
          </StyledThemeProvider>
        )
        .toJSON()
    ).toMatchSnapshot();
  });
  it("should render Button component", () => {
    const { getByTestId } = render(
      <StyledThemeProvider theme={theme}>
        <Button variant="filled" onClick={() => undefined} type="button">
          {children}
        </Button>
      </StyledThemeProvider>
    );
    const buttonComponent = getByTestId("button");
    expect(buttonComponent).toBeInTheDocument();
  });
});
