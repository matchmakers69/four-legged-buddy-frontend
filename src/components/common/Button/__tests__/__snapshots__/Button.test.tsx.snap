// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> sholud create a snapshot 1`] = `
<button
  className="Buttonstyled__Button-sc-13jglys-0 fQvjvz btn btn--danger"
  data-testid="button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div />
</button>
`;
